catalyst-gateway/bin/src/service/api/cardano/staking/assets_get.rs

```diff
@@ -111,25 +129,21 @@ struct TxoInfo {
 /// This function also updates the spent column if it detects that a TXO was spent
 /// between lookups.
 async fn calculate_stake_info(
-    persistent: bool, stake_address: Cip19StakeAddress, slot_num: Option<SlotNumber>,
+    session: &CassandraSession, stake_address: Cip19StakeAddress, slot_num: Option<SlotNo>,
 ) -> anyhow::Result<Option<StakeInfo>> {
-    let Some(session) = CassandraSession::get(persistent) else {
-        anyhow::bail!("Failed to acquire db session");
-    };
-
     let address: StakeAddress = stake_address.try_into()?;
     let stake_address_bytes = address.payload().as_hash().to_vec();
 
-    let mut txos_by_txn = get_txo_by_txn(&session, stake_address_bytes.clone(), slot_num).await?;
+    let mut txos_by_txn = get_txo_by_txn(session, stake_address_bytes.clone(), slot_num).await?;
     if txos_by_txn.is_empty() {
         return Ok(None);
     }
 
-    check_and_set_spent(&session, &mut txos_by_txn).await?;
+    check_and_set_spent(session, &mut txos_by_txn).await?;
     // TODO: This could be executed in the background, it does not actually matter if it
```

Comment:
This can't be a TODO, its critical we do this in the background otherwsie we will negatively impact performance for something we don't need to wait on.

Reply:
Fixing this will require a quite non trivial update, going to make it in the separate PR